src/xdsl/__init__.py

```diff
@@ -0,0 +1,38 @@
+_mlir_module = None
+_REQUIRED_MLIR_MODULES = ['ir']
+
+import types as _xdsl_init_types
```

Comment:
We usually used imports from `typing` for type hints in xdsl and not from `types`. I am not 100% sure what the difference is and I did not find a `ModuleType` in `typing`, but maybe someone knows better? @math-fehr

Reply:
I only looked what `importlib` uses TBH but, indeed, `typing` doesn't seem to have it. Also, [this SO post](https://stackoverflow.com/a/48389207/651937) points to `types`.

Reply:
Yeah, `types` is the right import here.
In Python, `typing` is for type hints, but `types` contains all the useful types that can then be used in the type hints (It's confusing I agree).